gold-plugins/two-step-auth.js

```diff
@@ -111,41 +131,90 @@ exports.commands = {
 		setup: function (target, room, user) {
 			if (!user.named) return this.errorReply("You must be logged in to use this command.");
 			if (!user.registered) return this.errorReply("You cannot setup two-step authentication on an account that isn't registered.");
+			if (Gold.userData[user.userid] && (Gold.userData[user.userid].email || Gold.userData[user.userid].twostepauth)) return this.errorReply("This account already has two-step authentication enabled.");
 			if (!target) return this.parse('/help twostep');
-			if (!target.includes('@')) return this.errorReply("This is not a valid email address.");
-			user.twostepEmail = {
-				email: target,
-				code: Math.floor(Math.random() * 90000) + 10000,
-			};
-			let email = `Hello, ${user.name}:\n\nTo verify this email account as a second step of authentication for your login on Gold, type this: /twostep  verify ${user.twostepEmail.code}`;
-			Gold.TwoStepAuth.verifyEmail(user, {email: target, message: email});
-			return this.sendReply("Check your email for verification - it will have you enter a command to verify that you own this email.");
+			let targets = target.split(',');
+			targets[0] = toId(targets[0]);
+			if (targets[0] !== 'email' && targets[0] !== 'authenticator' || (targets[0] === 'email' && !targets[1])) return this.parse('/help twostep');
+
+			if (targets[0] === 'email') {
+				if (!targets[1].includes('@')) return this.errorReply("This is not a valid email address.");
+				user.twostepEmail = {
+					email: targets[1],
+					code: Math.floor(Math.random() * 90000) + 10000,
+				};
+				let email = `Hello, ${user.name}:\n\nTo verify this email account as a second step of authentication for your login on Gold, type this: /twostep  verify ${user.twostepEmail.code}`;
+				Gold.TwoStepAuth.verifyEmail(user, {email: targets[1], message: email});
+				return this.sendReply("Check your email for verification - it will have you enter a command to verify that you own this email.");
+			} else if (targets[0] === 'authenticator') {
+				let twoAuthData = twoFactor.generateSecret({name: 'Gold PS', account: user.userid});
+				user.tempTwoAuth = twoAuthData.secret;
+				let uri = "otpauth://totp/Gold-PS:" + user.userid + "?secret=" + twoAuthData.secret + "&issuer=Gold-PS";
+				let qrImg = "https://chart.googleapis.com/chart?chs=166x166&chld=L|0&cht=qr&chl=" + uri;
+				let reply = "|modal||html|Please enter the following code into your authenticator application or scan the QR code.<br />";
+				reply += "Key: " + twoAuthData.secret + "<br />";
+				reply += "<img src=\"" + qrImg + "\" height=\"166\" width=\"166\"><br />";
+				reply += "Once you have added the key to your authenticator, please verify it by running ";
+				reply += "the following command:<br />";
+				reply += "<code>/twostep verify [code from your authenticator]</code>";
+				return user.popup(reply);
+			}
```

Comment:
There should be a fallout here telling the user their options are either email or authentication, and maybe sending them the command's syntax